### PR TITLE
fix: add support for endpoint prefix

### LINKED ISSUE #114: Endpoints prefix is not configurable through application.properties

### PR DESCRIPTION
**Pull request by mcollovati**
Adds a code generator that create the connect-client.ts file if a custom endpoint prefix has been configured in application.properties.

Fixes #114

### CODE REVIEW

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java [hunk 1]
```diff
@@ -284,12 +284,26 @@ void registerEndpoints(
...
                 QuarkusAtmosphereServlet.class.getName(), QuarkusAtmosphereServlet.class.getName());
-        builder.addMapping("/HILLA/push")
+        String prefix = endpointConfiguration.getEndpointPrefix();
+        if (prefix.matches("^/?connect/?$")) {
+            prefix = "/";
+        } else if (!prefix.startsWith("/")) {
+            prefix = "/" + prefix;
+        }
+        if (prefix.endsWith("/")) {
+            prefix = prefix.substring(0, prefix.length() - 1);
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:300]: Is the case relevant, when multiple `/`are present at the beginning or ending?

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java [hunk 2]
```diff
@@ -284,12 +284,26 @@ void registerEndpoints(
     }
 
     @BuildStep
-    void registerHillaPushServlet(BuildProducer<ServletBuildItem> servletProducer, NativeConfig nativeConfig) {
+    void registerHillaPushServlet(
+            BuildProducer<ServletBuildItem> servletProducer,
+            QuarkusEndpointConfiguration endpointConfiguration,
+            NativeConfig nativeConfig) {
         ServletBuildItem.Builder builder = ServletBuildItem.builder(
                 QuarkusAtmosphereServlet.class.getName(), QuarkusAtmosphereServlet.class.getName());
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:292]: Same name twice? Don't know the signature yet.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java [hunk 3]
```diff
@@ -284,12 +284,26 @@ void registerEndpoints(
...
+        }
+        if (prefix.endsWith("/")) {
+            prefix = prefix.substring(0, prefix.length() - 1);
+        }
+        String hillaPushMapping = prefix + "/HILLA/push";
+
+        builder.addMapping(hillaPushMapping)
                 .setAsyncSupported(true)
-                .addInitParam(ApplicationConfig.JSR356_MAPPING_PATH, "/HILLA/push")
+                .addInitParam(ApplicationConfig.JSR356_MAPPING_PATH, hillaPushMapping)
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:306]: Whats the difference between addMapping and addInitParam, are both required?

File: integration-tests/custom-prefix-test/src/main/resources/META-INF/resources/icons/icon.png
Comment (Dudeplayz) [inline integration-tests/custom-prefix-test/src/main/resources/META-INF/resources/icons/icon.png:1]: Icon wanted?

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        Path frontendDir = moduleDir.toPath().resolve(Path.of("src", "main", FRONTEND_FOLDER_NAME));
+        if (Files.isDirectory(frontendDir)) {
+            this.frontendFolder = frontendDir;
+            LOGGER.debug("Using frontend folder {}", frontendDir);
+        } else if (Files.isDirectory(legacyFrontendFolder)) {
+            this.frontendFolder = legacyFrontendFolder;
+            LOGGER.debug("Using legacy frontend folder {}", legacyFrontendFolder);
+        } else {
+            LOGGER.debug("Frontend folder not detected");
+        }
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:69]: Maybe throw some error or log some warning, if both folders are found. For example when missed to delete the old one, so both could be possible, which is an unwanted / unsupported case.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        Path legacyFrontendFolder = moduleDir.toPath().resolve(FRONTEND_FOLDER_NAME);
+        Path frontendDir = moduleDir.toPath().resolve(Path.of("src", "main", FRONTEND_FOLDER_NAME));
+        if (Files.isDirectory(frontendDir)) {
+            this.frontendFolder = frontendDir;
+            LOGGER.debug("Using frontend folder {}", frontendDir);
+        } else if (Files.isDirectory(legacyFrontendFolder)) {
+            this.frontendFolder = legacyFrontendFolder;
+            LOGGER.debug("Using legacy frontend folder {}", legacyFrontendFolder);
+        } else {
+            LOGGER.debug("Frontend folder not detected");
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:68]:
```suggestion
            LOGGER.debug("Frontend folder not found");
```
Just my preferred naming.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+                }
+            }
+        } else if (!defaultPrefix) {
+            return writeConnectClient(prefix, customClient);
+        }
+        return false;
+    }
+
+    private static String computeConnectClientPrefix(Config config) {
+        String prefix = config.getValue("vaadin.endpoint.prefix", String.class);
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:140]: Make the config key static and reuse it where possible or use an config interface for it and inject it.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                    LOGGER.debug("Template file connect-client.default.template.ts is empty.");
+                }
+            } catch (IOException ex) {
+                LOGGER.debug("Cannot read template file connect-client.default.template.ts.", ex);
+            }
+        } else {
+            LOGGER.debug("Cannot find template file connect-client.default.template.ts.");
+        }
+        return false;
+    }
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:170]: I would suggest guard clauses here for better readability.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+            try {
+                content = Files.readString(customClient);
+            } catch (IOException e) {
+                LOGGER.debug(
+                        "Cannot read content of custom connect-client.ts ({}). File will not be overwritten.",
+                        customClient);
+            }
+            if (content != null) {
+                content = content.replaceFirst(
+                        "(const client = new ConnectClient\\(\\{prefix:\\s*')[^']+('}\\);)", "$1" + prefix + "$2");
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:123]: Would only filter starting for `new ConnectClient(...)` (without `;`), a custom client could use another naming or add some trailing stuff in the line.

Comment (Dudeplayz) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:123]: Maybe also define the regex somewhere static

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        boolean defaultPrefix = "connect".equals(prefix);
+        Path customClient = context.inputDir().resolve("connect-client.ts");
+        if (Files.exists(customClient)) {
+            String content = null;
+            try {
+                content = Files.readString(customClient);
+            } catch (IOException e) {
+                LOGGER.debug(
+                        "Cannot read content of custom connect-client.ts ({}). File will not be overwritten.",
+                        customClient);
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:119]: Log the exception, so the actual reason for the exception can be investigated.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+            if (content != null) {
+                content = content.replaceFirst(
+                        "(const client = new ConnectClient\\(\\{prefix:\\s*')[^']+('}\\);)", "$1" + prefix + "$2");
+                try {
+                    Files.writeString(customClient, content);
+                    LOGGER.debug(
+                            "Prefix in custom connect-client.ts ({}) replaced with new value {}", customClient, prefix);
+                    return true;
+                } catch (IOException e) {
+                    LOGGER.debug("Cannot write content of custom connect-client.ts ({}).", customClient);
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:130]: Same here for logging the exception.

Comment (Dudeplayz) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:130]: Would also classify it as warn.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+        if (!Files.isDirectory(sourceDir)) {
+            return false;
+        }
+        String prefix = computeConnectClientPrefix(config);
+        boolean defaultPrefix = "connect".equals(prefix);
+        Path customClient = sourceDir.resolve("connect-client.ts");
+        if (Files.exists(customClient)) {
+            try {
+                String content = Files.readString(customClient);
+                if (!content.contains("const client = new ConnectClient({prefix: '" + prefix + "'});")) {
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:88]: Use the regex from later to be a bit more flexible?
I am not sure if my assumption is correct, that the user could also modify the connect client and it gets patched. Or do we assume always a quarkus-hilla generated one ?

File: commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+
+            assertThat(provider.getInputDirectory()).isEqualTo(customFrontendFolder);
+        } finally {
+            if (originalValue != null) {
+                System.setProperty(FrontendUtils.PARAM_FRONTEND_DIR, originalValue);
+            } else {
+                System.clearProperty(FrontendUtils.PARAM_FRONTEND_DIR);
+            }
+        }
+    }
```
Comment (Dudeplayz) [inline commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java:124]: Whats not clear here, remove it or keep it?

File: commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+        TypescriptClientCodeGenProvider.writeConnectClient("my-prefix", frontendFolder.resolve("connect-client.ts"));
+        configProperties.put("vaadin.endpoint.prefix", "/connect");
+
+        CodeGenContext context =
+                new CodeGenContext(appModel, outDir, projectDir, frontendFolder, true, buildConfig(), false);
+        provider.trigger(context);
+
+        Path connectClient = frontendFolder.resolve("connect-client.ts");
+        assertThat(connectClient).isRegularFile().content().contains(connectClientSnippet("connect"));
+    }
```
Comment (Dudeplayz) [inline commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java:244]: Maybe refactor, due much copy/paste ?

File: commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+
+    @Test
+    void shouldRun_customPrefix_customClientWithDifferentPrefix_run() throws IOException {
+        Path path = projectDir.resolve("frontend");
+        Files.createDirectories(path);
+        TypescriptClientCodeGenProvider.writeConnectClient("my-prefix", path.resolve("connect-client.ts"));
+        configProperties.put("vaadin.endpoint.prefix", "/different-prefix");
+
+        assertThat(provider.shouldRun(path, buildConfig())).isTrue();
+    }
```
Comment (Dudeplayz) [inline commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java:194]: Also refactor into one method with signatore of connectClientPrefix and newPrefix (or similiar). And determine the codeGen based on the difference of both? just keep the assertions in each method ?

**Comment by Dudeplayz** [comment]
One thing: Is this feature somewhere documented? Do we have documented the config properties somewhere at all? Maybe we add them directly to the readme ?

**Comment by mcollovati** [comment]
Custom prefix is a Hilla feature, but it would be good to mention automatic generation in the Readme

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java (hunk 1 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:300]: Many / is potentially a configuration error. We can throw an exception

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java (hunk 2 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:292]: Can you please elaborate a bit? I'm not sure I understood the comment

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java (hunk 3 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:306]: addMapping is for the AtmosphereServlet, the other one for Jakarta Websocket. They should match to work correctly

File: integration-tests/custom-prefix-test/src/main/resources/META-INF/resources/icons/icon.png
Comment (mcollovati) [reply integration-tests/custom-prefix-test/src/main/resources/META-INF/resources/icons/icon.png:1]: What icon?

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java (hunk 4 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:69]: A warning makes sense to me. Good point

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java (hunk 6 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:140]: I'm not sure inject is available during code generation (it runs also as a maven goal before build step processing). But static constant makes sense.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java (hunk 7 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:170]: Can you please elaborate? Just to be sure I got your point

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java (hunk 8 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:123]: If the client doesn't match the template I would not make any change

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java (hunk 11 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:88]: The idea is to patch the file only if the client creation statement matches the template

File: commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java (hunk 12 above)
Comment (mcollovati) [reply commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java:124]: Since the code generation runs also separately from the application (own goal) I am not sure we should handle the case of system property setting frontend folder. It looks like a corner case, maybe just fix if someone complains

File: integration-tests/custom-prefix-test/src/main/resources/META-INF/resources/icons/icon.png
Comment (mcollovati) [reply integration-tests/custom-prefix-test/src/main/resources/META-INF/resources/icons/icon.png:1]: Ah, now I see. It was not visible from mobile app

Comment (mcollovati) [reply integration-tests/custom-prefix-test/src/main/resources/META-INF/resources/icons/icon.png:1]: It is the same as in other test modules, coming from the Hilla starter project.
I would keep it.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java (hunk 2 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:292]: Now I see. It is correct. The first is the name of the servlet, the second is the servlet class.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java (hunk 1 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:300]: I suggest not complicating things and just handling leading and trailing slash.

Comment (Dudeplayz) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java:300]: I think thats fine. As you mentioned would be configuration error.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java (hunk 11 above)
Comment (Dudeplayz) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:99]: Yea thats good. Idea was just to be more unrestrictive, but the question is, in how many cases that would happen.

File: commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java (hunk 13 above)
Comment (mcollovati) [reply commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java]: done

File: commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java (hunk 12 above)
Comment (Dudeplayz) [reply commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java]: I think thats fine. As we are im a quarkus domain, configuration is preferred to use other mechanisms. I am always a friend of supporting corner cases, but if the effort is to high and the propability to low, just keep it as it is.

**Comment by Dudeplayz** [comment]
> Custom prefix is a Hilla feature, but it would be good to mention automatic generation in the Readme

Generation yes. For the prefix, I meant that the possible configuration keys that are possible in the application.properties for quarkus-hilla are documented. Just as simple code block with comments e.g.
```properties
vaadin.endpoint.prefix #custom hilla prefix ...
```

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 15]
```diff
@@ -116,7 +127,8 @@ public boolean trigger(CodeGenContext context) {
             } catch (IOException e) {
                 LOGGER.debug(
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:128]: This also as warn?

File: README.md [hunk 16]
```diff
@@ -51,11 +51,18 @@ integration eliminates the need for code duplication and ensures tighter alignme
 more streamlined updates and improved stability. By leveraging the Vaadin Quarkus extension, users of `quarkus-hilla`
 will benefit from enhanced compatibility with future Vaadin features.
 
+### Custom Endpoint Prefix
+
+A custom endpoint prefix can be configured setting the `vaadin.endpoint.prefix` entry in `application.properties`.
+The extension will create a custom `connect-client.ts` file in the `frontend` folder, constructing the `ConnectClient`
+object with the configured prefix.
+The `connect-client.ts` already exists and does not match the Hilla default template, it will not be overwritten.
```
Comment (Dudeplayz) [inline README.md:59]:
```suggestion
If `connect-client.ts` exists and does not match the default Hilla template, it is not overwritten.
```

File: commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java (hunk 12 above)
Comment (mcollovati) [reply commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java]: I propose to keep the disabled test for a while, and if we do not get feedback, remove it in a future release

File: commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java (hunk 14 above)
Comment (mcollovati) [reply commons/deployment/src/test/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProviderTest.java]: The code has been refactored, maybe now it covers your suggestion.
Otherwise, can you make an example?

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java (hunk 7 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java]: done

File: README.md [hunk 17]
```diff
@@ -51,11 +51,18 @@ integration eliminates the need for code duplication and ensures tighter alignme
 more streamlined updates and improved stability. By leveraging the Vaadin Quarkus extension, users of `quarkus-hilla`
 will benefit from enhanced compatibility with future Vaadin features.
 
+### Custom Endpoint Prefix
+
+A custom endpoint prefix can be configured setting the `vaadin.endpoint.prefix` entry in `application.properties`.
+The extension will create a custom `connect-client.ts` file in the `frontend` folder, constructing the `ConnectClient`
+object with the configured prefix.
```
Comment (Dudeplayz) [inline README.md:58]:
```suggestion
A custom endpoint prefix can be configured by setting the `vaadin.endpoint.prefix` entry in `application.properties`. The extension will create a custom `connect-client.ts` file in the `frontend` folder and construct the `ConnectClient` object with the configured prefix.
```

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+    static boolean writeConnectClient(String prefix, Path customClient) {
+        InputStream template = Endpoint.class.getResourceAsStream("/connect-client.default.template.ts");
+        if (template == null) {
+            LOGGER.debug("Cannot find template file connect-client.default.template.ts.");
+            return false;
+        }
+        try (InputStream is = template;
+                Scanner scanner = new Scanner(is).useDelimiter("\\A")) {
+            if (!scanner.hasNext()) {
+                LOGGER.debug("Template file connect-client.default.template.ts is empty.");
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:171]:
```suggestion
                LOGGER.debug("Template file connect-client.default.template.ts is empty.");
                return false;
```
Missing return, would be fatal.

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 19]
```diff
@@ -0,0 +1,182 @@
...
+            }
+            String out = scanner.next().replace("{{PREFIX}}", prefix);
+
+            Files.writeString(customClient, out, StandardCharsets.UTF_8);
+            return true;
+        } catch (IOException ex) {
+            LOGGER.debug("Cannot read template file connect-client.default.template.ts.", ex);
+        }
+        return false;
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:180]:
```suggestion
            LOGGER.debug("Cannot read template file connect-client.default.template.ts.", ex);
            return false;
        }
```

Should be valid, as the try/catch always returns something. In my opinion more clean, as there is no "fall through" case

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+            return false;
+        }
+        try (InputStream is = template;
+                Scanner scanner = new Scanner(is).useDelimiter("\\A")) {
+            if (!scanner.hasNext()) {
+                LOGGER.debug("Template file connect-client.default.template.ts is empty.");
+            }
+            String out = scanner.next().replace("{{PREFIX}}", prefix);
+
+            Files.writeString(customClient, out, StandardCharsets.UTF_8);
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:175]:
```suggestion
            String out = scanner.next().replace("{{PREFIX}}", prefix);
            Files.writeString(customClient, out, StandardCharsets.UTF_8);
```

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java (hunk 19 above)
Comment (mcollovati) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:180]: Not a fan of return in catch block, but it's fine

**Review by Dudeplayz — DISMISSED** [review]

File: commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java [hunk 21]
```diff
@@ -0,0 +1,182 @@
...
+                Scanner scanner = new Scanner(is).useDelimiter("\\A")) {
+            if (!scanner.hasNext()) {
+                LOGGER.debug("Template file connect-client.default.template.ts is empty.");
+                return false;
+            }
+            String out = scanner.next().replace("{{PREFIX}}", prefix);
+            Files.writeString(customClient, out, StandardCharsets.UTF_8);
+            return true;
+        } catch (IOException ex) {
+            LOGGER.debug("Cannot read template file connect-client.default.template.ts.", ex);
```
Comment (Dudeplayz) [inline commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:178]:
```suggestion
            LOGGER.warn("Cannot read template file connect-client.default.template.ts.", ex);
```

Warning would be better, as this should never happen?

Comment (Dudeplayz) [reply commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/TypescriptClientCodeGenProvider.java:178]: Also the write could fail, which is not distinguished.

**Review by Dudeplayz — APPROVED** [review]
